tests/external_botocore/test_bedrock_chat_completion.py

```diff
@@ -154,3 +160,82 @@ def test_bedrock_chat_completion_outside_txn(set_trace_info, exercise_model):
+
+
+# Testing get_ai_message_ids:
+
+
+@reset_core_stats_engine()
+@background_task()
+def test_get_ai_message_ids_when_nr_message_ids_not_set():
+    message_ids = get_ai_message_ids("request-id-1")
+    assert message_ids == []
+
+
+@reset_core_stats_engine()
+def test_get_ai_message_ids_outside_transaction():
+    message_ids = get_ai_message_ids("request-id-1")
+    assert message_ids == []
+
+
+@reset_core_stats_engine()
+def test_get_ai_message_ids_bedrock_chat_completion_in_txn(
+    set_trace_info, exercise_model, expected_ai_message_ids
+):  # noqa: F811
+    @background_task()
+    def _test():
+        set_trace_info()
+        add_custom_attribute("conversation_id", "my-awesome-id")
+        exercise_model(prompt=_test_bedrock_chat_completion_prompt, temperature=0.7, max_tokens=100)
+
+        id = [key for key in expected_ai_message_ids.keys()][0]
+        expected_message_ids = [value for value in expected_ai_message_ids.values()][0]
+        if id == "bedrock_key":
+            message_ids = [m for m in get_ai_message_ids()]
+            for index, message_id_info in enumerate(message_ids):
+                expected_message_id_info = expected_message_ids[index]
+                assert message_id_info["conversation_id"] == expected_message_id_info["conversation_id"]
+                assert message_id_info["request_id"] == expected_message_id_info["request_id"]
+                assert message_id_info["message_id"]
+
+        else:
+            message_ids = [m for m in get_ai_message_ids(id)]
+            for index, message_id_info in enumerate(message_ids):
+                expected_message_id_info = expected_message_ids[index]
+                assert message_id_info["conversation_id"] == expected_message_id_info["conversation_id"]
+                assert message_id_info["request_id"] == expected_message_id_info["request_id"]
+                assert message_id_info["message_id"] == expected_message_id_info["message_id"]
+
+        assert current_transaction()._nr_message_ids == {}
+
+    _test()
+
+
+@reset_core_stats_engine()
+def test_get_ai_message_ids_bedrock_chat_completion_no_convo_id(
+    set_trace_info, exercise_model, expected_ai_message_ids
+):  # noqa: F811
+    @background_task()
+    def _test():
+        set_trace_info()
+        exercise_model(prompt=_test_bedrock_chat_completion_prompt, temperature=0.7, max_tokens=100)
+
+        id = [key for key in expected_ai_message_ids.keys()][0]
+        expected_message_ids = [value for value in expected_ai_message_ids.values()][0]
+        if id == "bedrock_key":
+            message_ids = [m for m in get_ai_message_ids()]
+            for index, message_id_info in enumerate(message_ids):
+                expected_message_id_info = expected_message_ids[index]
+                assert message_id_info["request_id"] == expected_message_id_info["request_id"]
+                assert message_id_info["message_id"]
```

Comment:
The conversation id is always returned from the get_ai_message_ids call, even if it isn't set (aka empty string).
```suggestion
                assert message_id_info["message_id"]
                assert message_id_info["conversation_id"] == ""
```